File: Courses/PHP Web Basics - май 2019/16.ExerciseEncapsulationInheritance/02.03.Mankind/Student.php
<?php

require_once "Human.php";

class Student extends Human
{
    /**
     * @var string
     */
    private $facultyNumber;
    public function __construct(string $firstName, string $lastName, string $facultyNumber)
    {
        parent::__construct($firstName, $lastName);
        $this->setFacultyNumber($facultyNumber);

    }

    /**
     * @return string
     */
    public function getFacultyNumber(): string
    {
        return $this->facultyNumber;
    }

    /**
     * @param string $facultyNumber
     * @throws Exception
     */
    private function setFacultyNumber(string $facultyNumber): void
    {
        if (strlen($facultyNumber) < 5 || strlen($facultyNumber) > 10) {
            throw new \Exception("Invalid faculty number!");
        }

        $this->facultyNumber = $facultyNumber;
    }

    /**
     * @return string
     */
    public function __toString()
    {
        return "First Name: " . $this->getFirstName().PHP_EOL."Last Name: " . $this->getLastName().PHP_EOL."Faculty number: " . $this->getFacultyNumber().PHP_EOL.PHP_EOL;
        //First Name: {student's first name}
	    //Last Name: {student's last name}
	    //Faculty number: {student's faculty number}

    }
}

File: Courses/C# OOP/Basics/10. Inheritance Exercise/Exercises_Inheritance/03.Mankind/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace _03.Mankind
{
    public class Student : Human
    {
        private string facultNumber;

        public Student(string firstName, string lastName, string facultNumber) : base(firstName, lastName)
        {
            FacultNumber = facultNumber;
        }

        public string FacultNumber
        {
            get => facultNumber; 
            set
            {
                if (value.Length < 5 || value.Length > 10 || !value.ToCharArray().All(char.IsLetterOrDigit))
                {
                    throw new ArgumentException("Invalid faculty number!");
                }
                facultNumber = value;
            }
        }

        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.AppendLine($"First Name: {FirstName}");
            sb.AppendLine($"Last Name: {LastName}");
            sb.AppendLine($"Faculty number: {FacultNumber}");
            return sb.ToString();
        }
    }
}
